egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,247 @@
+use egui::emath::RectTransform;
+use egui::epaint::{CircleShape, CubicBezierShape, QuadraticBezierShape};
+use egui::*;
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(default))]
+pub struct PaintBezier {
+    bezier: usize,                       // current bezier curve degree, it can be 3,4,
+    tolerance: f32,                      // the tolerance for the bezier curve
```

Comment:
```suggestion
```

no need for this here - just read it from `ctx.tessellation_options`. Never duplicate state if it can be avoided :)

Reply:
yes, agree and fixed. however, I can't find an elegant way to drive the option. I'm really unhappy with the code between lines 64 and 68. Appreciate a better idea.